### PR TITLE
[CY-9838] make client testable

### PR DESCRIPTION
**Pull request by yvgenycyolo**
make client testable
create interface in order to allow DI for client

### CODE REVIEW

File: ldap-client.go [hunk 1]
```diff
@@ -22,7 +22,19 @@ var (
 	// errUserNotIdentified represents a situation in which the user could not be identified.
 	errUserNotIdentified = fmt.Errorf("user could not be indentifed")
 )
-
+type LDAPClienter interface{
```
Comment (zalmen) [inline ldap-client.go:25]: add space after `interface`

File: ldap-client.go [hunk 2]
```diff
@@ -22,7 +22,19 @@ var (
...
+	Connect() error
+	Bind(dn,password string) error
+	Close()
+	UsersSearch(orFilter string,uidAttr string) (bool, map[string]map[string][]string, error)
+	RunQueries(username string, queries []string) (results map[string]bool, err error)
+	GetAllGroupsByName(groupName string) ([]*LdapGroup, error)
+	GetAllGroupsWithMembersByDN(groupDN []string) ([]*LdapGroup, error)
+	ChangeADUserPassword(username, oldPassword, newPassword string) (err error)
+    ChangeOpenLDAPUserPassword(username, oldPassword, newPassword string) (err error)
```
Comment (zalmen) [inline ldap-client.go:34]: fix alignment (or is it a github issue?)

File: ldap-client.go [hunk 3]
```diff
@@ -85,6 +97,12 @@ func (lc *LDAPClient) Connect() error {
 	return nil
 }
 
+func (lc *LDAPClient) Bind(dn,password string) error {
+	if err := lc.Conn.Bind(dn,password); err != nil {
+		return fmt.Errorf("LDAP: cannot bind to server: invalid credentials or bad DN")
```
Comment (zalmen) [inline ldap-client.go:102]: add the original `err` for better context

File: ldap-client.go [hunk 4]
```diff
@@ -438,6 +456,27 @@ func (lc *LDAPClient) ChangeOpenLDAPUserPassword(username, oldPassword, newPassw
 	return nil
 }
 
+func (lc *LDAPClient) GetUserByCN(userCN,uidAttr string)(uid string, err error){
```
Comment (zalmen) [inline ldap-client.go:459]: add missing spaces

File: ldap-client.go [hunk 5]
```diff
@@ -438,6 +456,27 @@ func (lc *LDAPClient) ChangeOpenLDAPUserPassword(username, oldPassword, newPassw
...
+		[]string{"cn", uidAttr},
+		nil,
+	)
+
+	sr, err := lc.Conn.Search(searchRequest)
+	if err != nil {
+		return "", err
+	}
+	if len(sr.Entries) != 1 {
+		return "", nil
```
Comment (zalmen) [inline ldap-client.go:473]: IIUC, it means that we found more than one user, right? 
If yes, and assuming it is an illegal behavior, you should return an error.

File: ldap-client.go [hunk 6]
```diff
@@ -438,6 +456,27 @@ func (lc *LDAPClient) ChangeOpenLDAPUserPassword(username, oldPassword, newPassw
...
+	}
+	userEntry := sr.Entries[0]
+	uid = userEntry.GetAttributeValue(uidAttr)
```
Comment (zalmen) [inline ldap-client.go:476]: can it fail? (I assume not, but double checking)

File: ldap-client.go (hunk 2 above)
Comment (yvgenycyolo) [reply ldap-client.go:34]: github issue, I see it ok

File: ldap-client.go (hunk 6 above)
Comment (yvgenycyolo) [reply ldap-client.go:476]: cannot fail

**Pull request merged**